### PR TITLE
feat: Add permission boundary to IAM role

### PR DESCRIPTION
**Pull request by topikachu**
## what
* add permissions_boundary to aws_iam_role

## why
* Our org requires all IAM role has permissions_boundary

## references
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role#permissions_boundary
* cloudposse/terraform-aws-eks-iam-role#29

### CODE REVIEW

**Comment by topikachu** [comment]
@nitrocode  Could you review this change too?
^-^

**Review by Gowiem — DISMISSED** [review]
LGTM

**Comment by topikachu** [comment]
@Gowiem  Can somebody merge this change?
Thanks

File: variables.tf [hunk 1]
```diff
@@ -338,3 +338,9 @@ variable "postrender_binary_path" {
   description = "Relative or full path to command binary."
   default     = null
 }
+
+variable "permissions_boundary" {
+  type        = string
+  description = "If provided, IAM role will be created with this permissions boundary attached"
```
Comment (nitrocode) [inline variables.tf:344]: Please use the same description

https://github.com/cloudposse/terraform-aws-eks-iam-role/blob/b87373712671560e774b9733004111795495c377/variables.tf#L46

Comment (topikachu) [reply variables.tf]: Fixed

**Review by Gowiem — APPROVED** [review]
LGTM

**Comment by Gowiem** [comment]
@topikachu our tests are running into the following.

```
TestExamplesComplete 2022-07-26T14:07:35Z logger.go:66: Running command terraform with args [apply -input=false -auto-approve -var example=Hello, world! -var attributes=["25062"] -var-file fixtures.us-east-2.tfvars -lock=false]
TestExamplesComplete 2022-07-26T14:07:37Z logger.go:66: 
TestExamplesComplete 2022-07-26T14:07:37Z logger.go:66: Error: Value for undeclared variable
TestExamplesComplete 2022-07-26T14:07:37Z logger.go:66: 
TestExamplesComplete 2022-07-26T14:07:37Z logger.go:66: A variable named "example" was assigned on the command line, but the root
TestExamplesComplete 2022-07-26T14:07:37Z logger.go:66: module does not declare a variable of that name. To use this value, add a
TestExamplesComplete 2022-07-26T14:07:37Z logger.go:66: "variable" block to the configuration.
TestExamplesComplete 2022-07-26T14:07:37Z logger.go:66: 
TestExamplesComplete 2022-07-26T14:07:37Z retry.go:80: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; 
Error: Value for undeclared variable
```

@nitrocode looks like these might using some default test code... did this repo never get setup with tests? Have we been merging against this repo without that?

**Comment by nitrocode** [comment]
Yes, the PR to add the tests are here https://github.com/cloudposse/terraform-aws-helm-release/pull/16 and are incomplete. I was blocked. If you or someone can help, that would be cool :)

**Pull request merged**